src/avitab/apps/PlaneManualApp.cpp

```diff
@@ -16,109 +16,27 @@
  *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 #include "PlaneManualApp.h"
-#include "src/avitab/apps/components/FileSelect.h"
-#include "src/avitab/apps/components/PDFViewer.h"
-#include "src/platform/Platform.h"
-#include "src/Logger.h"
 
 namespace avitab {
 
 PlaneManualApp::PlaneManualApp(FuncsPtr appFuncs):
-    App(appFuncs)
+    DocumentsApp(appFuncs, "Manuals", "manualsapp", "\\.(pdf|png|jpg|jpeg|bmp)$")
 {
-    showAircraftFolder();
+    browseStartDirectory = findStartDirectory();
+    Run();
 }
 
-void PlaneManualApp::show() {
-    auto aircraftPath = api().getAirplanePath();
-    if (aircraftPath != currentAircraft) {
-        errorMsg.reset();
-        childApp.reset();
-        showAircraftFolder();
-    }
-
-    if (childApp) {
-        childApp->show();
-    } else {
-        App::show();
-    }
-}
-
-void PlaneManualApp::showAircraftFolder() {
-    currentAircraft = api().getAirplanePath();
-    bool showError = false;
+std::string PlaneManualApp::findStartDirectory() {
+    auto basePath = api().getAirplanePath();
 
-    if (platform::fileExists(currentAircraft + "manual/")) {
-        currentPath = currentAircraft + "manual/";
-    } else if (platform::fileExists(currentAircraft + "manuals/")) {
-        currentPath = currentAircraft + "manuals/";
-    } else if (platform::fileExists(currentAircraft + "documentation/")) {
-        currentPath = currentAircraft + "documentation/";
-    } else {
-        showError = true;
-        if (platform::fileExists(currentAircraft)) {
-            currentPath = currentAircraft;
-        } else {
-            currentPath = api().getDataPath();
+    const char *search[] = { "manual", "manuals", "documentation", "docs", 0 };

```

Comment:
Please use `std::string` for this instead, and a range based for loop below.

Reply:
Done.